rerun_cpp/CMakeLists.txt

```diff
@@ -25,43 +30,14 @@ if(MSVC)
 endif()
 
 # ------------------------------------------------------------------------------
-# Rerun development setup only:
-if(DEFINED RERUN_REPOSITORY)
-    # TODO(andreas): use add_custom_command instead so this runs at build time! https://cmake.org/cmake/help/latest/command/add_custom_command.html#command:add_custom_command
-    execute_process(COMMAND cargo build --release -p rerun_c RESULT_VARIABLE ret) # We link against this, so must be up-to-date
-
-    # execute process doesn't fail if the process fails.
-    # `COMMAND_ERROR_IS_FATAL ANY` parameter fixes this but is only available in CMake 3.19
-    if(NOT(ret EQUAL "0"))
-        message(FATAL_ERROR "Failed to build rerun_c.")
-    endif()
-
-    # Overwrite where to find rerun_c library.
-    if(APPLE)
-        set(RERUN_C_LIB_DEFAULT ${CMAKE_CURRENT_SOURCE_DIR}/../target/release/librerun_c.a)
-    elseif(UNIX) # if(LINUX) # CMake 3.25
-        set(RERUN_C_LIB_DEFAULT ${CMAKE_CURRENT_SOURCE_DIR}/../target/release/librerun_c.a)
-    elseif(WIN32)
-        set(RERUN_C_LIB_DEFAULT ${CMAKE_CURRENT_SOURCE_DIR}/../target/release/rerun_c.lib)
-    else()
-        message(FATAL_ERROR "Unsupported platform.")
-    endif()
-
-    # Set very strict warning settings when we're testing the SDK.
-    # We don't want to force this on any user!
-    set_default_warning_settings(rerun_sdk)
-
-    # Put `rerun.h` into the same place where it's on a user's machine and apply CMake variables like version number.
-    configure_file(
-        "${CMAKE_CURRENT_SOURCE_DIR}/../crates/rerun_c/src/rerun.h"
-        "${CMAKE_CURRENT_SOURCE_DIR}/src/rerun/c/rerun.h"
-        NEWLINE_STYLE LF # Specify line endings, otherwise CMake wants to change them on Windows.
-    )
-endif()
+# Setup rerun_c dependency.
+add_library(rerun_c STATIC IMPORTED)
 
-# ------------------------------------------------------------------------------
-# Setup rerun_c dependency:
-if(NOT DEFINED RERUN_C_LIB_DEFAULT)
+# Inside the repo build ourselves, otherwise default to a local `lib` folder.
+if(DEFINED RERUN_REPOSITORY)
```

Comment:
Is it possible to do something like call `build_rerun_c` from the root-level CMakeLists.txt and then pass it in by setting `RERUN_C_LIB`? That way in isolation this CMakeLists.txt wouldn't need to be aware of `RERUN_REPOSITORY`.

Reply:
Moving out the build_rerun_c is definitely possible, what I'm less sure about is this issue about having to add the target - just adding the file doesn't work because then cmake doesn't know that it has to build the target that produces the file 🤦 . But maybe we can do something like `if (DEFINED RERUN_C_TARGET)` instead and if that's defined use that rerun_c target, otherwise set it up ourselves 🤔

Reply:
ah I think I found a nice way to structure this and have no more `RERUN_REPOSITORY` checks in here!

Reply:
https://github.com/rerun-io/rerun/pull/4126/commits/1604522bb882443ed2d61ae9420cb1b790802abf